packages/styled-system/src/pseudos.ts

```diff
@@ -221,8 +221,26 @@ export const pseudoSelectors = {
    * this component or element.
    */
   _dark: ".dark &, [data-theme=dark] &",
+  /**
+   * Styles for when `.light` is applied to any parent of
+   * this component or element.
+   */
+  _light: ".light &, [data-theme=light] &",
 }
 
 export type Pseudos = typeof pseudoSelectors
 
+export const getPrefixedPseudoSelectors = (theme: any) => {
+  const prefix = theme?.config?.cssVarPrefix ?? ""
+  return {
+    ...pseudoSelectors,
+    _dark: `.${applyPrefix("dark", prefix)} &, [data-theme=dark] &`,
+    _light: `.${applyPrefix("light", prefix)} &, [data-theme=light] &`,
+  }
+}
```

Comment:
I thought it might be nice to let `_dark` target a custom name like `.my-app-dark`, but idk if this is really necessary.

For now I leveraged `cssVarPrefix` so it defaults  to `chakra-dark`. Previously it was `chakra-ui-dark` so this would actually be a breaking change. If we wanted to keep it backwards compatible we could just add a new property to `theme.config` like `colorModePrefix` instead of using `cssVarPrefix`.